### PR TITLE
Improve description for MTRDevice.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Includes various information that wasn't included before but probably should be.

This includes #34856 to avoid merge conflicts.

### CODE REVIEW

**Review by jtung-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 1]
```diff
@@ -543,8 +587,58 @@ - (void)dealloc
...
+        wifi = @"NO";
+        thread = @"NO";
+    } else {
+        wifi = YES_NO(networkFeatures.unsignedLongLongValue & MTRNetworkCommissioningFeatureWiFiNetworkInterface);
+        thread = YES_NO(networkFeatures.unsignedLongLongValue & MTRNetworkCommissioningFeatureThreadNetworkInterface);
+    }
+
+    NSString * reportAge;
+    if (mostRecentReportTime) {
+        reportAge = [NSString stringWithFormat:@" (%.0fs ago)", -[mostRecentReportTime timeIntervalSinceNow]];
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:628]:
```suggestion
        reportAge = [NSString stringWithFormat:@" (%.0lfs ago)", -[mostRecentReportTime timeIntervalSinceNow]];
```

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 2]
```diff
@@ -543,8 +587,58 @@ - (void)dealloc
...
+    } else {
+        reportAge = @"";
+    }
+
+    NSString * subscriptionFailureAge;
+    if (lastSubscriptionFailureTime) {
+        subscriptionFailureAge = [NSString stringWithFormat:@" (%.0fs ago)", -[lastSubscriptionFailureTime timeIntervalSinceNow]];
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:635]:
```suggestion
        subscriptionFailureAge = [NSString stringWithFormat:@" (%.0lfs ago)", -[lastSubscriptionFailureTime timeIntervalSinceNow]];
```

Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTRDevice.mm]: Done.

File: src/darwin/Framework/CHIP/MTRDevice.mm (hunk 1 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTRDevice.mm]: Done.

**Pull request merged**